python/cudf/cudf/core/index.py

```diff
@@ -2744,6 +2748,214 @@ def categories(self):
+        If None, dtype will be inferred.
+
+    copy : bool, default False
+        Copy the input data.
+
+    name : object, optional
+        Name to be stored in the index.
+
+    Returns
+    -------
+    IntervalIndex
+    """
+
+    def __new__(
+        cls, data=None, closed=None, dtype=None, copy=False, name=None,
+    ) -> "IntervalIndex":
+        if copy:
+            data = column.as_column(data, dtype=dtype).copy(deep=True)
+        out = Frame.__new__(cls)
+        kwargs = _setdefault_name(data, name=name)
+        if isinstance(data, IntervalColumn):
+            data = data
+        elif isinstance(data, pd.Series) and (is_interval_dtype(data.dtype)):
+            data = column.as_column(pa.Array.from_pandas(data), dtype=dtype,)
+        elif isinstance(data, (pd._libs.interval.Interval, pd.IntervalIndex)):
+            data = column.as_column(data, dtype=dtype,)
+        else:
+            if data is not None and data != [] and data[0].closed != closed:
+                # when closed is not the same as the data's closed
+                # we need to change the data closed value,
+                left_col = [data[i].left for i in range(len(data))]
+                # creating a col out of the left child so we can get
+                # the correct dtype
+                left = column.as_column(left_col)
+                data = column.as_column(
+                    data,
+                    dtype=IntervalDtype(left.dtype, closed=closed)
+                    if dtype is None
+                    else dtype,
+                )
+            if not data and closed != "right":
+                data = column.build_interval_column([], [], closed=closed)
+            else:
+                data = column.as_column(
+                    data, dtype="interval" if dtype is None else dtype
+                )
+
+        out._initialize(data, **kwargs)
+        return out
+
+    def from_breaks(breaks, closed="right", name=None, copy=False, dtype=None):
+        """
+        Construct an IntervalIndex from an array of splits.
+
+        Parameters
+        ---------
+
+        breaks : array-like (1-dimensional)
+            Left and right bounds for each interval.
+
+        closed : {"left", "right", "both", "neither"}, default "right"
+            Whether the intervals are closed on the left-side, right-side,
+            both or neither.
+
+        Returns
+        -------
+        IntervalIndex
+
+        Examples
+        --------
+        >>> import cudf
+        >>> import pandas as pd
+        >>> cudf.IntervalIndex.from_breaks([0, 1, 2, 3])
+        IntervalIndex([(0, 1], (1, 2], (2, 3]],
+                    closed='right',
+                    dtype='interval[int64]')
+        """
+
+        left_col = breaks[:-1:]
+        right_col = breaks[+1::]
+
+        interval_col = column.build_interval_column(
+            left_col, right_col, closed=closed
+        )
+
+        return IntervalIndex(interval_col)
```

Comment:
@kkraus14  I also thought I'd raise that IntervalIndex doesn't have `get_item` support in libcudf. This makes the code a bit messier down the line with `cut` because `libcudf::labels_bin` provides the correct indexes but I can't actually index into an IntervalIndex. I'm creating a new array with individual IntervalIndexes but raising in case we can work on it in the future.
Adding an example below for more context. 

```

In [5]: pd_interval[0]
Out[5]: Interval(0, 5, closed='left')

In [6]: cudf_interval = cudf.interval_range(start=0,end=10, periods=2,closed='left')

In [7]: cudf_interval[0]
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-7-0d267651b90a> in <module>
----> 1 cudf_interval[0]

~/cudf/python/cudf/cudf/core/index.py in __getitem__(self, index)
   2109 
   2110     def __getitem__(self, index):
-> 2111         res = self._values[index]
   2112         if not isinstance(index, int):
   2113             res = as_index(res)

~/cudf/python/cudf/cudf/core/column/column.py in __getitem__(self, arg)
    646     def __getitem__(self, arg) -> Union[ScalarLike, ColumnBase]:
    647         if is_scalar(arg):
--> 648             return self.element_indexing(int(arg))
    649         elif isinstance(arg, slice):
    650             start, stop, stride = arg.indices(len(self))

~/cudf/python/cudf/cudf/core/column/column.py in element_indexing(self, index)
    624             raise IndexError("single positional indexer is out-of-bounds")
    625 
--> 626         return libcudf.copying.get_element(self, idx).value
    627 
    628     def slice(self, start: int, stop: int, stride: int = None) -> ColumnBase:

~/cudf/python/cudf/cudf/_lib/copying.pyx in cudf._lib.copying.get_element()

RuntimeError: cuDF failure at: ../../../../src/copying/get_element.cu:166: get_element_functor not supported for struct_view
```

Reply:
Yea, struct scalars aren't implemented yet so this is to be expected. Ideally we could throw a nice `NotImplementedError` here if possible.

Reply:
Ok, sounds good! I'll do that :D